task04/test/unit/public-holidays.service.test.ts

```diff
@@ -0,0 +1,109 @@
+import axios from 'axios';
+import {
+  getListOfPublicHolidays,
+  checkIfTodayIsPublicHoliday,
+  getNextPublicHolidays
+} from '../../services/public-holidays.service';
+import {PUBLIC_HOLIDAYS_API_URL, SUPPORTED_COUNTRIES} from '../../config';
+import {holidays, shortHolidays} from '../test-data';
+
+const year = 2024;
+const country = SUPPORTED_COUNTRIES[0];
+
+describe('Get list of public holidays', () => {
+  test('should return list of short holidays',  async () => {
+    jest.spyOn(axios, 'get').mockImplementation(() => Promise.resolve({ data: holidays }));
```

Comment:
I would move the axios library mock to a higher level (e.g. to the `beforeAll` level, where I would lock all necessary methods), and `mockImplementation` would already be defined at the level of specific tests.

Reply:
Fixed